sigopt/validate/experiment_input.py

```diff
@@ -138,6 +138,8 @@ def validate_experiment_input(experiment_input):
       'The project field is not permitted in the experiment.'
       ' Please set the SIGOPT_PROJECT environment variable instead.'
     )
+  if RUNS_ONLY_KEY in experiment_input:
+    raise ValidationError(f"The {RUNS_ONLY_KEY} field is not allowed for experiments created with this module.")
```

Comment:
Error message might be a little misleading, as we do call this method right before adding the runs-only key for sigopt-factory - is it just that users shouldn't really be supplying this themselves?

Reply:
Why is it misleading? It is not allowed/supported with this module.

Reply:
I guess I don't really know what we mean by `module` in this case - is that a term we're using elsewhere? Do we expect someone to be calling this code directly, rather than through the factory / magics helpers?

Like, what is the user supposed to do with this information if they think they want to include this info - what are their other options for `modules`?

Reply:
This is referring to the `sigopt` module, aka sigopt python client.

> Like, what is the user supposed to do with this information if they think they want to include this info
This is not intended as a publicly documented field. I would consider it's manual use to be erroneous for the time being, and the user should exclude it.